tests/protocol/test_connectors.py

```diff
@@ -1974,3 +1974,75 @@ def test_pipeline_properties(key, value, default_value):
 )
 def test_get_advanced_rules(filtering, expected_advanced_rules):
     assert Filter(filtering).get_advanced_rules() == expected_advanced_rules
+
+
+def test_updated_configuration():
+    current = {
+        "tenant_id": {"label": "Tenant ID", "order": 1, "type": "str", "value": "foo"},
+        "tenant_name": {
+            "label": "Tenant name",
+            "order": 2,
+            "type": "str",
+            "value": "bar",
+        },
+        "client_id": {"label": "Client ID", "order": 3, "type": "str", "value": "baz"},
+        "secret_value": {
+            "label": "Secret value",
+            "order": 4,
+            "sensitive": True,
+            "type": "str",
+            "value": "qux",
+        },
+        "some_toggle": {"label": "toggle", "order": 5, "type": "bool", "value": False},
+    }
+    simple_default = {
+        "tenant_id": {
+            "label": "Tenant Identifier",
+            "order": 1,
+            "type": "str",
+        },
+        "tenant_name": {
+            "label": "Tenant name",
+            "order": 2,
+            "type": "str",
+        },
+        "new_config": {
+            "label": "Config label",
+            "order": 3,
+            "type": "bool",
+            "value": True,
+        },
+        "client_id": {
+            "label": "Client ID",
+            "order": 4,
+            "type": "str",
+        },
+        "secret_value": {
+            "label": "Secret value",
+            "order": 5,
+            "sensitive": True,
+            "type": "str",
+        },
+        "some_toggle": {"label": "toggle", "order": 6, "type": "bool", "value": True},
+    }
+    missing_configs = ["new_config"]
+    connector = Connector(elastic_index=Mock(), doc_source={"_id": "test"})
+    result = connector.updated_configuration(missing_configs, current, simple_default)
+
+    # all keys included where there are changes (excludes 'tenant_name')
```

Comment:
One question, in protocol/connector.py we have:
```
doc["configuration"] = self.updated_configuration(
                missing_keys, current_config, simple_config
            )
```
(it might be because it's friday afternoon) I don't fully understand why the `tenant_name` is expected to be missing since it's included in current config and simple_default config in the test case

Would it result in the actual connector doc miss that field in the index?

Reply:
> I don't fully understand why the tenant_name is expected to be missing since it's included in current config and simple_default config in the test case

The two are identical in all fields, so there's no need to include it in the partial update request, since it wouldn't lead to a change.

> Would it result in the actual connector doc miss that field in the index?

I think this is the gap/misunderstanding. We're using the `_update` elasticsearch API here, which does partial updates. You actually can't remove objects/keys with this API, you can only add new keys and edit existing leaf values.